katsdptelstate/aio/test/test_telescope_state.py

```diff
@@ -31,520 +33,512 @@
+            await ts.delete('test_key')
             task = asyncio.ensure_future(set_key())
-            with async_timeout.timeout(2):
-                await self.ts.wait_indexed('test_key', 'bar', condition)
-            self.assertEqual(2, await self.ts.get_indexed('test_key', 'bar'))
+            async with async_timeout.timeout(2):
+                await ts.wait_indexed('test_key', 'bar', condition)
+            assert 2 == await ts.get_indexed('test_key', 'bar')
             await task
 
-    async def test_wait_indexed_cancel(self) -> None:
-        task = asyncio.ensure_future(self.ts.wait_indexed('test_key', 'sub_key'))
+    async def test_wait_indexed_cancel(self, ts: TelescopeState) -> None:
+        task = asyncio.ensure_future(ts.wait_indexed('test_key', 'sub_key'))
         await asyncio.sleep(0.1)
         task.cancel()
-        with self.assertRaises(asyncio.CancelledError):
+        with pytest.raises(asyncio.CancelledError):
             await task
 
-    async def test_wait_indexed_shadow(self) -> None:
+    async def test_wait_indexed_shadow(self, ts: TelescopeState, ns: TelescopeState) -> None:
         async def set_key(telstate):
             await asyncio.sleep(0.1)
             await telstate.set_indexed('test_key', 'sub_key', 1)
 
         # Shadow the key that will be set by set_key
-        await self.ns.set_indexed('test_key', 'blah', 1)
-        task = asyncio.ensure_future(set_key(self.ts))
-        with self.assertRaises(asyncio.TimeoutError):
-            with async_timeout.timeout(0.2):
-                await self.ns.wait_indexed('test_key', 'sub_key')
+        await ns.set_indexed('test_key', 'blah', 1)
+        task = asyncio.ensure_future(set_key(ts))
+        with pytest.raises(asyncio.TimeoutError):
+            async with async_timeout.timeout(0.2):
+                await ns.wait_indexed('test_key', 'sub_key')
         await task
 
-    async def test_wait_indexed_wrong_type(self) -> None:
+    async def test_wait_indexed_wrong_type(self, ts: TelescopeState) -> None:
         async def set_key():
             await asyncio.sleep(0.1)
-            await self.ts.add('test_key', 'value')
+            await ts.add('test_key', 'value')
 
         task = asyncio.ensure_future(set_key())
-        with self.assertRaises(ImmutableKeyError):
-            await self.ts.wait_indexed('test_key', 'value')
+        with pytest.raises(ImmutableKeyError):
+            await ts.wait_indexed('test_key', 'value')
         await task
         # Different code-path when key was already set at the start
-        with self.assertRaises(ImmutableKeyError):
-            await self.ts.wait_indexed('test_key', 'value')
+        with pytest.raises(ImmutableKeyError):
+            await ts.wait_indexed('test_key', 'value')
 
-    async def _test_mixed_unicode_bytes(self, ns, key) -> None:
-        await self.ts.clear()
+    async def _test_mixed_unicode_bytes(self, ts: TelescopeState, ns: TelescopeState, key) -> None:
```

Comment:
`key: str`?

Reply:
That was my first reaction, but it's tested with a mix of `str` and `bytes`. I could make it a Union I suppose.